src/ziggurat/streams.clj

```diff
@@ -63,15 +61,16 @@
 (defn- map-values [mapper-fn stream-builder]
   (.mapValues stream-builder (value-mapper mapper-fn)))
 
-(defn- transformer-supplier [metric-namespace oldest-processed-message-in-s]
+(defn- transformer-supplier
+  [metric-namespace oldest-processed-message-in-s topic-entity-name]
   (reify TransformerSupplier
-    (get [_] (transformer/create metric-namespace oldest-processed-message-in-s))))
+    (get [_] (transformer/create metric-namespace oldest-processed-message-in-s topic-entity-name))))
 
-(defn- transform-values [topic-entity oldest-processed-message-in-s stream-builder]
-  (let [metric-namespace (get-metric-namespace "message-received-delay-histogram" topic-entity)]
-    (.transform stream-builder (transformer-supplier metric-namespace oldest-processed-message-in-s) (into-array [(.name (store-supplier-builder))]))))
+(defn- transform-values [topic-entity-name oldest-processed-message-in-s stream-builder]
+  (let [metric-namespace "message-received-delay-histogram"]
```

Comment:
metric-name change will not be backward compatible.

Reply:
done, the call to (mk-histogram) added another one using the new metric-namesapce